pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -228,6 +228,25 @@ stdenv.mkDerivation ({
         configureFlags+=" --extra-lib-dirs=$p/lib"
       fi
     done
+
+    if "${if stdenv.isDarwin then "true" else "false"}"; then
```

Comment:
Considering the fact you *have* conditionals in the Nix language, why is that feature used here to create input for yet another level of conditional that's evaluated by the shell? This looks awkward. Is there a deeper reason for this implementation choice?